NightscoutServiceKit/NightscoutService.swift

```diff
@@ -375,17 +368,149 @@ extension NightscoutService: RemoteDataService {
         })
     }
     
-    enum NotificationValidationError: LocalizedError {
-        case missingOTP
+    public func handleRemoteNotification(_ notification: [String: AnyObject]) async throws {
+        let commandSource = try commandSource(notification: notification)
+        await commandSource.handleRemoteNotification(notification)
+    }
+    
+    public func processPendingRemoteCommands() async throws {
+        try await commandSourceV2.processPendingRemoteCommands()
+    }
+    
+    private func commandSource(notification: [String: AnyObject]) throws -> RemoteCommandSource {
         
-        var errorDescription: String? {
-            switch self {
-            case .missingOTP:
-                return "Error: Password is required."
-            }
+        guard let versionString = notification["version"] as? String else {
+            return commandSourceV1 //fallback before versions were introduced
+        }
+        
+        guard let version = Double(versionString) else {
+            throw NightscoutServiceError.missingCommandSource
+        }
+        
+        switch version {
+        case 1.0..<2.0:
+            return commandSourceV1
+        case 2.0..<3.0:
+            return commandSourceV2
+        default:
+            throw NightscoutServiceError.missingCommandSource
+        }
+    }
+
+}
+
+extension NightscoutService: RemoteCommandSourceV1Delegate {
+    
+    func commandSourceV1(_: RemoteCommandSourceV1, handleAction action: Action) async throws {
+        
+        switch action {
+        case .temporaryScheduleOverride(let overrideCommand):
+            try await self.serviceDelegate?.handleRemoteOverride(
+                name: overrideCommand.name,
+                durationTime: overrideCommand.durationTime,
+                remoteAddress: overrideCommand.remoteAddress
+            )
+        case .cancelTemporaryOverride:
+            try await self.serviceDelegate?.handleRemoteOverrideCancel()
+        case .bolusEntry(let bolusCommand):
+            try await self.serviceDelegate?.handleRemoteBolus(amountInUnits: bolusCommand.amountInUnits)
+        case .carbsEntry(let carbCommand):
+            try await self.serviceDelegate?.handleRemoteCarb(
+                amountInGrams: carbCommand.amountInGrams,
+                absorptionTime: carbCommand.absorptionTime,
+                foodType: carbCommand.foodType,
+                startDate: carbCommand.startDate
+            )
+        case .autobolus(let autobolusCommand):
+            try await self.serviceDelegate?.handleRemoteAutobolus(activate: autobolusCommand.active)
+        case .closedLoop(let closedLoopComand):
+            try await self.serviceDelegate?.handleRemoteClosedLoop(activate: closedLoopComand.active)
         }
     }
+    
```

Comment:
This will upload the NS note. A screenshot of the result is in the main Loop PR.